app/src/androidTest/java/com/github/h3lp3rs/h3lp/AwaitHelpActivityTest.kt

```diff
@@ -0,0 +1,267 @@
+        bundle.putStringArrayList(EXTRA_NEEDED_MEDICATION, arrayListOf(EPIPEN))
+        val intent = Intent(
+            getApplicationContext(),
+            AwaitHelpActivity::class.java
+        ).apply {
+            putExtras(bundle)
+        }
+        return launch(intent)
+    }
+
+    private fun launchAndDo(popup: Boolean, action: () -> Unit) {
+        launch(popup).use {
+            init()
+            action()
+            release()
+        }
+    }
+
+    @Test
+    fun callEmergenciesButtonWorksAndSendIntent() {
+        launchAndDo(false) {
+
+            val phoneButton = onView(withId(R.id.await_help_call_button))
+
+            phoneButton.inRoot(RootMatchers.isFocusable()).perform(click())
+
+            // click the ambulance in the popup
+            onView(withId(R.id.ambulance_call_button)).inRoot(RootMatchers.isFocusable())
+                .perform(click())
+
+            intended(
+                allOf(
+                    hasAction(ACTION_DIAL)
+                )
+            )
+        }
+    }
+
+    @Test
+    fun showsPopUpIfNotCalledBeforeAndCanCallEmergencies() {
+        launchAndDo(true) {
+
+            val phonePopupButton = onView(withId(R.id.open_call_popup_button))
+            phonePopupButton.inRoot(RootMatchers.isFocusable()).perform(click())
+
+            intended(
+                allOf(
+                    hasAction(ACTION_DIAL)
+                )
+            )
+        }
+    }
+
+//    @Test
+//    fun clickPhoneButtonAndContactButtonDialsEmergencyContactNumber() {
+//        launchAndDo(false) {
+//
+//            // Clicking on the call for emergency button
+//            val phoneButton = onView(withId(R.id.await_help_call_button))
+//
+//            //phoneButton.inRoot(RootMatchers.isFocusable()).check(matches(isDisplayed()))
+//            phoneButton.inRoot(RootMatchers.isFocusable()).perform(click())
+//
+//            // click the contact button in the popup
+//            onView(withId(R.id.contact_call_button)).inRoot(RootMatchers.isFocusable())
+//                .perform(click())
+//
+//            // The expected ambulance phone number given the location (specified by the coordinates)
+//            val number = "tel:$VALID_CONTACT_NUMBER"
+//
+//            // Checking that this emergency number is dialed
+//            intended(
+//                allOf(
+//                    hasAction(ACTION_DIAL) //,
+//                    // hasData(Uri.parse(number)) This poses a problem for Cirrus
+//                )
+//            )
+//        }
+//    }
+
+    private fun clickingOnButtonWorksAndSendsIntent(
+        ActivityName: Class<*>?,
+        id: Matcher<View>,
+        isInScrollView: Boolean
+    ) {
+        launchAndDo(false) {
+
+            if (isInScrollView) {
+                onView(id).inRoot(RootMatchers.isFocusable()).perform(click())
+            } else {
+                onView(id).inRoot(RootMatchers.isFocusable()).perform(click())
+            }
```

Comment:
Avoiding a little code duplication
```suggestion
            val viewInteraction = onView(id).inRoot(RootMatchers.isFocusable())
            if (isInScrollView) {
                viewInteraction.perform(click())
            } else {
                viewInteraction.perform(click())
            }

```